docs/resources/sequence.md

```diff
@@ -42,12 +42,13 @@ resource "snowflake_sequence" "test_sequence" {
 
 - `comment` (String) Specifies a comment for the sequence.
 - `increment` (Number) The amount the sequence will increase by each time it is used
+- `ordering` (String) The ordering of the sequence. Either ORDER or NOORDER. Default is ORDER.
```

Comment:
Do we leave ORDER as a default in the provider? I mean, it changes in https://docs.snowflake.com/en/release-notes/bcr-bundles/2024_01/bcr-1483 to NOORDER by default.

So the current behavior would be aligned with Snowflake but it will change in the couple of days. Introducing NOORDER as a default would break existing configs (not break but cause different behavior) but it will be ready for the change later.

Maybe the one other way would be to do a small breaking change in the provider by introducing the ordering as a required param (adding it to our [migration notes](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md)) and letting users to decide themselves? WDYT @sfc-gh-swinkler @sfc-gh-jcieslak.

Reply:
It's a small behavior change, maybe the best would be to stay with the previous default to not break the existing configs and have a task to change it (to the manual setting or different default) as a pre-v1 change.

Reply:
we can just make this field nullable. if it is null, it will not set it, and therefore use whatever behavior is default

Reply:
okay so i made the change, but I'm still not sure I am happy with it. feels janky. The problem is that during the read I only want to set the attribute if it was set in the config (otherwise it will look like drift .... user applies, then will be forever stuck in a planning loop until they edit their Terraform configuration). However, during the import there is no way of knowing whether we should set the configuration code or not.How to know if value is default or not default? I think this is an important design question we should answer, as there are many places where we do similar operations. So we should all agree on a general solution

Reply:
i do know in some places (e.g. warehouses) we cannot rely on the default value in the docs, either because the default is wrong for that particular account, or because it could be setting feature flags that are only available for business critical, and therefore wouldn't work in other accounts. So having a null value for an attribute is a necessity. but understanding when to set this during the read is a mystery.

Reply:
Discussed during the call:
- we rollback these changes
- we stick with explicitly stating our default
- we may change it with V1